staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -281,16 +287,27 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 
 	<-stopCh
 
-	return s.RunPreShutdownHooks()
+	err = s.RunPreShutdownHooks()
+	if err != nil {
+		return err
+	}
+
+	s.HandlerChainWaitGroup.Wait()
```

Comment:
please add a comment above that we wait for requests to finish, and that is bounded by the RequestTime variable.

Reply:
ok